cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -45,6 +46,90 @@ Wants=%[1]s
 	doSystemdMount = doSystemdMountImpl
 )
 
+// forbiddenChars is a list of characters that are not allowed in any mount paths used in systemd-mount.
+const forbiddenChars = `\,:" `
+
+// overlayFsOptions groups the options to systemd-mount related to overlayfs.
+type overlayFsOptions struct {
+	// Directories to be used as lower layers of an overlay mount.
+	// It does not need to be on a writable filesystem.
+	LowerDirs []string
+	// A directory to be used as the upper layer of an overlay mount.
+	// This is normally on a writable filesystem.
+	UpperDir string
```

Comment:
Technically upper can be absent so perhaps leave a case where the empty string is "we don't select upper at all".

Reply:
Thanks for the review! Indeed upperdir and workdir can be empty so I tweaked my checks in Validate.

Reply:
Thank you